### PR TITLE
Support backtickeds on index level

### PR DESCRIPTION
**Pull request by katejim**

### CODE REVIEW

**Comment by niktrop** [comment]
I would suggest to change some method names in ScalaNamesUtil:
`convertMemberName` -> `clean`
`convertMemberFqn` -> `cleanFqn`
`fqnNamesEquals` -> `equivalentFqn`
`memberNamesEquals` -> `equivalent`
`addBacktickedIfScalaKeywordFqn` -> `escapeKeywordsFqn`
`changeKeyword` -> `escapeKeyword`

But with such names these methods should always be used with `ScalaNamesUtil` qualifier.

File: src/org/jetbrains/plugins/scala/lang/refactoring/util/ScalaNamesUtil.scala [hunk 1]
```diff
@@ -96,8 +103,40 @@ object ScalaNamesUtil {
...
+  def fqnNamesEquals(l: String, r: String): Boolean = {
+    if (l == r) return true
+    convertMemberFqn(l) == convertMemberFqn(r)
   }
+
+  def memberNamesEquals(l: String, r: String): Boolean = {
+    if (l == r) return true
+    convertMemberName(l) == convertMemberName(r)
+  }
+
```
Comment (niktrop) [inline src/org/jetbrains/plugins/scala/lang/refactoring/util/ScalaNamesUtil.scala]: It seems that these two methods are not needed anymore.

File: src/org/jetbrains/plugins/scala/lang/refactoring/util/ScalaNamesUtil.scala [hunk 2]
```diff
@@ -96,8 +103,40 @@ object ScalaNamesUtil {
...
+    val toDecode = name match {
+      case ScalaNamesUtil.isBacktickedName(s) => s
+      case _ => name
+    }
+    NameTransformer.decode(toDecode)
+  }
+
+  def convertMemberFqn(fqn: String): String =
+    splitName(fqn).map(convertMemberName).mkString(".")
+
```
Comment (niktrop) [inline src/org/jetbrains/plugins/scala/lang/refactoring/util/ScalaNamesUtil.scala:116]: `l == r || convertMemberFqn(l) == convertMemberFqn(r)`

**Comment by katejim** [comment]
merged manually

**Pull request closed without merging**